sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2038,12 +2038,13 @@ block_store(AccessorT acc, simd<T, N> vals, simd_mask<1> pred,
 // Implementations of accessor-based gather and scatter functions
 namespace detail {
 template <typename T, int N, typename AccessorTy>
-ESIMD_INLINE
-    ESIMD_NODEBUG std::enable_if_t<(sizeof(T) <= 4) &&
-                                   (N == 1 || N == 8 || N == 16 || N == 32) &&
-                                   !std::is_pointer_v<AccessorTy>>
-    scatter_impl(AccessorTy acc, simd<T, N> vals, simd<uint32_t, N> offsets,
-                 uint32_t glob_offset, simd_mask<N> mask) {
+ESIMD_INLINE ESIMD_NODEBUG std::enable_if_t<
+    (sizeof(T) <= 4) && (N == 1 || N == 8 || N == 16 || N == 32) &&
+    (std::is_same_v<detail::LocalAccessorMarker, AccessorTy> ||
+     detail::is_accessor_with_v<AccessorTy,
```

Comment:
minor comment: `detail::` is redundant here. `scatter_impl` function is already inside `detail` namespace.